lightning/src/ln/features.rs

```diff
@@ -395,6 +410,77 @@ impl InvoiceFeatures {
 	}
 }
 
+impl ToBase32 for InvoiceFeatures {
+	fn write_base32<W: WriteBase32>(&self, writer: &mut W) -> Result<(), <W as WriteBase32>::Err> {
+		(&self.flags[..]).write_base32(writer)
+	}
+}
+
+impl Base32Len for InvoiceFeatures {
+	fn base32_len(&self) -> usize {
+		bytes_size_to_base32_size(self.flags.len())
+	}
+}
+
+impl FromBase32 for InvoiceFeatures {
+	type Err = bech32::Error;
+
+	fn from_base32(field_data: &[u5]) -> Result<InvoiceFeatures, bech32::Error> {
+		let mut features = InvoiceFeatures::empty();
```

Comment:
Features can be of arbitrary length, so we really need to just read the bytes, I think, not check individual bits and copy them.

Reply:
Hmm so sadly, `Vec::<u8>::from_base32(vec_of_u5s)` isn't working for me. E.g., if a bit is in the 9th to rightmost position in the vec(u5), it won't still be in the 9th-to-rightmost position after the conversion. 

It's making me think I need to write a custom converter like this Stackoverflow answer? https://stackoverflow.com/a/7135008 Does that sound right?

Reply:
Hmm, that's definitely what `Vec::<u8>::from_base32` should already be doing - https://docs.rs/bech32/0.8.0/src/bech32/lib.rs.html#238-240

Reply:
Yeah, related to offline discussion, it turns out I just need to bit shift the results :)

Reply:
So, shifting the bits ended up being very edge casey and I spent too much time on it, and it ended up being much easier to just copy over the bits manually, and not use the [given Vec<u8> implementations](https://docs.rs/bech32/0.8.0/bech32/trait.ToBase32.html#impl-ToBase32)